### ISSUE TITLE
Skip data replay history before agents show up

### ISSUE THREAD

**Issue by liamchzh** [huawei-noah/SMARTS#160]
Here is what the training script looks like:
```
for agent_id, mission in agent_missions.items():
    scenario.set_ego_missions({agent_id: mission})
    smarts = SMARTS(...)
    observations = smarts.reset(scenario)
    while not dones[agent_id]:
        ...
````

In the agent's mission, a `start_time` will be specified to indicate when the agent will show up. Suppose the agent shows up at t1, and `smarts.reset` will step through from t0 to t1 (Basically just replay the history data), which is not necessary. To speed up the training, we can ignore the history between t0-t1 and start at t1.

@zbzhu99

**Comment by Gamenot** [comment]
@Adaickalavan The main focus with this is the TrafficHistoryProvider. 

We will define:
* t0 as 0 or the first step of the simulation (whichever is greater)
* t1 as the first time an agent is in the scenario.

In a scenario.py (e.g. 4lane_t/scenario.py) we can add `start_time=50` which means that the agent will enter at a time of 50 seconds into the simulation.

The current way that smarts resets is it will complete all the steps up to the first time an agent is in the simulation (50 seconds into the simulation). For data replay we want to skip that time by skipping the traffic histories to t1 but starting smarts simulation at t0.

With that we can play the traffic histories starting from t1 which maps to a smarts simulation time of t0.

**Comment by JingfeiPeng** [comment]
closed with #596
